core/store.go

```diff
@@ -63,3 +64,20 @@ func Del(k string) bool {
 	}
 	return false
 }
+
+func FilterVals(pattern string) [][2]string {
+	var matchedValues [][2]string
+	for key, value := range store {
+		val := value.Value.(string)
+		ok, err := filepath.Match(pattern, val)
+		if err != nil {
+			return nil
+		}
+		if ok {
+			matchedValues = append(matchedValues, [2]string{key, value.Value.(string)})
+		}
+		value.LastAccessedAt = getCurrentClock()
+	}
+
+	return matchedValues
```

Comment:
Also can't we convert the [][2]string to []string here itself?

Reply:
by string, you mean []string? because we'll have a list of matches right?

Reply:
@manosriram the return value should be an array to two-tuple allowing we to pipe it with different commands (in the future)

Reply:
noted, updated in 537614751

Reply:
@manosriram Since the requirement is to return an array of two-tuple so it can be piped to another set of commands. The something on these lines maybe-

```go 

type filtervalTuple struct {
    Rkey string; 
    Rvalue interface{}
}

func FilterVals(pattern string) []filtervalTuple {
    rdevice := regexp.MustCompile(pattern)
    var result []filtervalTuple

    for key, value := range store {
        val := value.(string)
        ok := rdevice.FindString(val)

        if ok !="" {
            result = append(result, filtervalTuple{Rkey:key, Rvalue:val})
        }
    }

    return result
}

```
Check a sample code here! - https://rohanverma.godbolt.org/z/5f43zrMTo

@arpitbbhayani any further input if you can give then it would be helpful for @manosriram

Reply:
@rohanverma94 not sure if the regex will match wildcards, updated other parts except that.

Reply:
@manosriram It's just a sample code for simplifying the concept :smiley:

Reply:
@rohanverma94 okay 😅